### PR TITLE
boards/arm/samv7: rework linker script usage logic to allow custom linker scripts

### PR DESCRIPTION
**Pull request by pkarashchenko**
## Summary
This PR breaks SAMV7 dependency on `BOARD_COMMON_DIR` for linker script generation logic:
- Allow to locate linker script templates at specific board level instead of board common level
- Generate kernel build linker script from a template
- Move make rules from common to per-board `Make.defs`

## Impact
Introduce an option for custom boards to have custom board based linker scripts

## Testing
Pass CI

### CODE REVIEW

File: boards/arm/samv7/common/scripts/kernel-space.ld [hunk 1]
```diff
@@ -18,7 +18,7 @@
  *
  ****************************************************************************/
 
-/* NOTE:  This depends on the memory.ld script having been included prior to
+/* NOTE:  This depends on the memory.ld.template script having been included prior to
```
Comment (xiaoxiang781216) [inline boards/arm/samv7/common/scripts/kernel-space.ld:21]: move "prior to" to next line

File: boards/arm/samv7/common/scripts/user-space.ld [hunk 2]
```diff
@@ -18,7 +18,7 @@
  *
  ****************************************************************************/
 
-/* NOTE:  This depends on the memory.ld script having been included prior to
+/* NOTE:  This depends on the memory.ld.template script having been included prior to
```
Comment (xiaoxiang781216) [inline boards/arm/samv7/common/scripts/user-space.ld:21]: ditto

File: boards/arm/samv7/common/scripts/kernel-space.ld (hunk 1 above)
Comment (pkarashchenko) [reply boards/arm/samv7/common/scripts/kernel-space.ld]: done

File: boards/arm/samv7/common/scripts/user-space.ld (hunk 2 above)
Comment (pkarashchenko) [reply boards/arm/samv7/common/scripts/user-space.ld]: done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**